be/src/exprs/array_functions.tpp

```diff
@@ -1079,6 +1087,105 @@ public:
+
+        return dest_column;
+    }
+
+    static void _sort_multi_array_column(FunctionContext* ctx, const Column& src_array_column,
+                                         const std::vector<ColumnPtr>& key_array_columns,
+                                         Column* dest_array_column) {
+        const auto& src_elements_column = down_cast<const ArrayColumn&>(src_array_column).elements();
+        const auto& src_offsets_column = down_cast<const ArrayColumn&>(src_array_column).offsets();
+
+        auto* dest_elements_column = down_cast<ArrayColumn*>(dest_array_column)->elements_column().get();
+        auto* dest_offsets_column = down_cast<ArrayColumn*>(dest_array_column)->offsets_column().get();
+        dest_offsets_column->get_data() = src_offsets_column.get_data();
+
+        auto src_elements_size = src_elements_column.size();
+        std::vector<uint32_t> key_sort_index;
+        key_sort_index.reserve(src_elements_size);
+
+        size_t chunk_size = src_array_column.size();
+        size_t key_array_size = key_array_columns.size();
+
+        std::vector<ColumnPtr> key_elements_columns;
+        std::vector<std::shared_ptr<UInt32Column>> key_offsets_columns;
+        for (size_t i = 0; i < key_array_size; ++i) {
+            ColumnPtr key_data = key_array_columns[i];
+            if (key_data->is_nullable()) {
+                const auto* key_nullable_column = down_cast<const NullableColumn*>(key_data.get());
+                key_data = key_nullable_column->data_column();
+            }
+
+            const auto& key_element_column = down_cast<ArrayColumn*>(key_data.get())->elements_column();
+            const auto& key_offset_column = down_cast<ArrayColumn*>(key_data.get())->offsets_column();
+
+            key_elements_columns.push_back(key_element_column);
+            key_offsets_columns.push_back(key_offset_column);
+        }
+
+        const std::atomic<bool>& cancel = ctx->state()->cancelled_ref();
+        Status status;
+        SortDescs sort_desc = SortDescs::asc_null_first(src_elements_size);
+
+        for (size_t i = 0; i < chunk_size; ++i) {
+            auto src_column_offset_size = src_offsets_column.get_data()[i + 1] - src_offsets_column.get_data()[i];
+            for (size_t key_column_size = 0; key_column_size < key_array_size; ++key_column_size) {
+                auto key_column_offset_size = key_offsets_columns[key_column_size].get()->get_data()[i + 1] -
+                                              key_offsets_columns[key_column_size]->get_data()[i];
+                if (src_column_offset_size != key_column_offset_size) {
+                    throw std::runtime_error("Input arrays' size are not equal in array_sortby.");
+                }
+            }
+
+            Permutation perm;
+            auto range = std::make_pair(src_offsets_column.get_data()[i], src_offsets_column.get_data()[i + 1]);
+            status = sort_and_tie_columns(cancel, key_elements_columns, sort_desc, &perm, range);
```

Comment:
If `key_nullable_column->immutable_null_column_data[i]` is true, we should skip this key column for the row `i`.

For example,  `SELECT array_sortby([1,2,3], [3, 2, 2], null, [1,2,3])` returns `[2,3,1]`, which is the same as `SELECT array_sortby([1,2,3], [3, 2, 2], [1,2,3])`